internal/config/profile.go

```diff
@@ -457,24 +457,25 @@ func (p *Profile) IsTelemetryEnabledSet() bool {
 // TelemetryEnabled get the configured telemetry enabled value.
 func TelemetryEnabled() bool { return Default().TelemetryEnabled() }
 func (p *Profile) TelemetryEnabled() bool {
-	if ToolName != AtlasCLI || !isTelemetryFeatureFlagSet() {
-		return false
-	}
-	return p.GetBool(telemetryEnabled)
+	return isTelemetryFeatureAllowed() && p.GetBool(telemetryEnabled)
 }
 
 // SetTelemetryEnabled sets the telemetry enabled value.
 func SetTelemetryEnabled(v bool) { Default().SetTelemetryEnabled(v) }
 func (p *Profile) SetTelemetryEnabled(v bool) {
-	if ToolName != AtlasCLI || !isTelemetryFeatureFlagSet() {
+	if !isTelemetryFeatureAllowed() {
 		return
 	}
 	SetGlobal(telemetryEnabled, v)
 }
 
-func isTelemetryFeatureFlagSet() bool {
-	_, featureFlagSet := os.LookupEnv("MONGODB_ATLAS_TELEMETRY_FEATURE_FLAG")
-	return featureFlagSet
+func boolEnv(key string) bool {
+	value, ok := os.LookupEnv(key)
+	return ok && IsTrue(value)
+}
+
+func isTelemetryFeatureAllowed() bool {
+	return ToolName == AtlasCLI && !boolEnv("DO_NOT_TRACK") && boolEnv(AtlasCLIEnvPrefix+"_TELEMETRY_FEATURE_FLAG")
```

Comment:
Viper will add automatically the prefix to the env variable https://github.com/spf13/viper#working-with-environment-variables. I am not sure how  `doNotTrack`  is used but all the env variable should have the `MONGODB_ATLAS_CLI` prefix
```suggestion
	return ToolName == AtlasCLI && !viper.GetBool(doNotTrack) && viper.GetBool(telemetryFeatureFlag")
```

Reply:
@andreaangiolillo the problem of using viper here is it will check the configuration file as well, and these two properties are not suppose to be in the config at all.

DO_NOT_TRACK is more of a standard so no prefix for it.

Reply:
> these two properties are not supposed to be in the config at all

ah okay. I don't understand why. Could you clarify? so far, all the env vars have been added to the conf file, so this decision seems a little strange to me. I don't understand how checking the conf file may be a problem 🤔  the conf file is not going to be changed manually by customers (we are not documenting it) so we can assume these prop will not be in the conf file.

Reply:
We have a telemetry_enable configuration. That comes from file/env var.

Telemetry_Feature_Flag is a temporary measure created by @johansteyn so we don't track customers now, we are suppose to remove it. If we use config file we will end up with an entry that serves no purpose.

DO_NOT_TRACK is not atlascli env var, it is a standard documented in https://consoledonottrack.com which will not have a config file entry nor prefix in it.

Reply:
Thanks for the clarification! : )

Reply:
@andreaangiolillo the ticket has a link to :https://consoledonottrack.com/

It's a generic env var that the industry seems to be adopting to disable tracking across the board, ie. it is not specific to any particular apps.